### PR TITLE
Use standard unix domain socket from dart:io

### PR DESCRIPTION
**Pull request by robert-ancell**

### CODE REVIEW

**Comment by robert-ancell** [comment]
We should shut down https://github.com/CanonicalLtd/unix_domain_socket.dart after this is landed (delete or update description to indicate is obsolete?)

File: lib/src/dbus_client.dart [hunk 1]
```diff
@@ -43,59 +77,102 @@ class DBusClient {
...
-    _messageStream.listen((dynamic receivedData) {
-      var message = receivedData as DBusMessage;
-      if (message.type == MessageType.Signal)
-        onSignal(
-            message.path, message.interface, message.member, message.values);
-    });
+  _authenticate() async {
+    // Send an empty byte, as this is required if sending the credentials as a socket control message.
+    // We rely on the server using SO_PEERCRED to check out credentials.
+    _socket.add([ 0 ]);
```
Comment (MarcusTomlinson) [inline lib/src/dbus_client.dart:127]: The Dart formatting standard requires this to be `_socket.add([0]);`

File: lib/src/dbus_client.dart [hunk 2]
```diff
@@ -1,9 +1,8 @@
-import 'package:unix_domain_socket/unix_domain_socket.dart';
-
 import "dart:async";
 import "dart:convert";
```
Comment (MarcusTomlinson) [inline lib/src/dbus_client.dart:2]: Unused import.

File: lib/src/dbus_read_buffer.dart [hunk 3]
```diff
@@ -23,6 +24,17 @@ class DBusReadBuffer {
...
+  String readLine() {
+    for (var i = readOffset; i < data.length - 1; i++) {
+      if (data[i] == 13 /* '\r' */ && data[i + 1] == 10 /* '\n' */) {
+        var bytes = List<int>(i - readOffset);
+        for (var j = readOffset; j < i; j++) bytes[j] = readByte();
+        readOffset = i + 2;
+        return utf8.decode(bytes);
+      }
+    }
+  }
```
Comment (MarcusTomlinson) [inline lib/src/dbus_read_buffer.dart:36]: This method is missing a fallback `return` if no `\r` or `\n` is found.

**Review by MarcusTomlinson — CHANGES_REQUESTED** [review]
This is awesome Robert! Just a few small nits.

**Comment by MarcusTomlinson** [comment]
Also, the example apps don't close when they reach the end of `main()`. Is this due to https://github.com/CanonicalLtd/dbus_client.dart/issues/2?

I noticed you reference this PR on that issue, was this PR meant to fix it?

**Comment by MarcusTomlinson** [comment]
> We should shut down https://github.com/CanonicalLtd/unix_domain_socket.dart after this is landed (delete or update description to indicate is obsolete?)

Yeah in fact I'm gonna merge this now, archive that project, and mark the pub.dev package "discontinued".

I'll just commit the few review fixes myself.

**Pull request merged**